pkg/common/utils.go

```diff
@@ -172,3 +172,16 @@ func GetSharedServicesNamespace(ctx context.Context, name string) (namespace str
 	}
 	return
 }
+
+func MergeMap(in map[string]string, mergeMap map[string]string) map[string]string {
+	if in == nil {
+		in = make(map[string]string)
+	}
+	if mergeMap == nil {
+		mergeMap = make(map[string]string)
```

Comment:
Shouldn't we just return in the event that `mergeMap == nil`? It's essentially a no-op on `in`, which we know at this point is at least an empty `map[string]string`.

Reply:
hi @rwhundley , Nil check for `in` map is not really required, as from the MergeMap function calls, i ensured `in` map always has data, I just added nil check for `in` map too thinking in future if someone wants to consume MergeMap function we validate `in` is not nil.

Reply:
Right, but we should be coding defensively here against misuse. What we don't want are situations where binaries panic. That said, the way this is written now will be fine given we are still making empty `map[string]string` for both vars. I was just saying that, if one is non-`nil` and the other is `nil`, we could just return whichever value has been submitted (a no-op.)